### PR TITLE
Clean up Exception macro doxygen.

### PR DESCRIPTION
**Pull request by chrisdembia**
Primarily, I'm making sure these macros show up on the page for the Exception class.

You should be able to view the resulting doxygen at http://myosin.sourceforge.net/732/classOpenSim_1_1Exception.html.

### CODE REVIEW

File: OpenSim/Common/Exception.h [hunk 1]
```diff
@@ -50,36 +50,50 @@
...
 
 
-/** Macros to throw OpenSim exceptions. The purpose of these macros is to avoid
-having to provide the first few arguments (which are common to all OpenSim
-exceptions) to the exception constructor in the throw statements. This also
-allows us to add more details (eg class name) later easily.                   */
+/** @name Macros to throw OpenSim exceptions
+The purpose of these macros is to avoid having to provide the first few
+arguments (which are common to all OpenSim exceptions) to the exception
+constructor in the throw statements.
```
Comment (sherm1) [inline OpenSim/Common/Exception.h]: I think the purposes, in order of decreasing importance are:
- ensure consistency of message formatting
- ensure that all messages contain file/line/func information
- make it easy for developers to produce good messages
- save typing

**Comment by sherm1** [comment]
I looked at the doxygen. It looks good, but strangely lists "Frank C Anderson" as the author! It might be time for an update ...

**Comment by chrisdembia** [comment]
I addressed both of your comments.

**Comment by sherm1** [comment]
:+1: ready to merge unless you want to wait for the builds.

**Comment by chrisdembia** [comment]
Thanks. Appveyor failed for other reasons, so I'm merging.

**Pull request merged**